packages/server/src/integrations/base/sql.ts

```diff
@@ -89,6 +89,24 @@ function parseFilters(filters: SearchFilters | undefined): SearchFilters {
   return filters
 }
 
+function generateSelectStatement(json: QueryJson, knex: Knex): any[] {
```

Comment:
Can we be more specific than `any` for the type here? I believe this is either a string or a knex raw statement, but this should have a type and it would make it more obvious what this is doing.

I think knex raw may actually return a string, if this is a casting issue/typescript being unaware, you can do something like `return knex.raw(`....`) as string`

Reply:
Knex.raw returns a 'Knew.Raw' type. I tried casting as a string / using `.toSQL().sql` but that didn't work.

I realised afterwards anyway that when you pass a string into the `select` method, it does the client specific column wrapping, e.g. for Postgres it wraps the column name in double quotes.

This was a problem because it would return `"TableName.ColumnName::money::numeric"` 
You need the raw type to tell knex not to do any additional quote wrapping to the string.

Reply:
Ah, in that case we can just `(string | Knex.Raw)[]` to define an array of mixed types - its helpful in the SQL layer to try to apply types where possible to these statements, there's quite a bit of complexity and it'll help us catch any weirdness at build time (since Knex is already pretty well typed).

Reply:
I didn't know you could do that - cool! 

That's done. In the process, I also removed a couple of unused parameters in `addRelationships` just so you're aware.